search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/prompt/PromptUtil.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2023 Aryn
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.opensearch.searchpipelines.questionanswering.generative.prompt;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import org.apache.commons.text.StringEscapeUtils;
+
+import java.util.List;
+import java.util.Locale;
+
+/**
+ * TODO Should prompt engineering llm-specific?
+ *
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class PromptUtil {
+
+    private static final String roleUser = "user";
+
+    public static String getQuestionRephrasingPrompt(String originalQuestion, List<String> chatHistory) {
+        return null;
+    }
+
+    public static String getChatCompletionPrompt(String question, List<String> chatHistory, List<String> contexts) {
+        return buildMessageParameter(question, chatHistory, contexts);
+    }
+
+    @VisibleForTesting
+    static String buildMessageParameter(String question, List<String> chatHistory, List<String> contexts) {
```

Comment:
I understand we will build better prompt template later. For now, you can use following i think if you want to reduce the token counts:

```
"Craft a succinct, under 100-word response considering:"
+ "- Numbered search results"
+ "- Rephrased initial query"
+ "- Chat history"
+ "Reference sources as [${number}]."
+ "Avoid repetition, even from history."
+ "Include steps if needed."
```

Reply:
We are hesitant to switch the prompt until we test it out first.  The current prompt has been tested on a few datasets including some customer data.

Reply:
We should definitely address the token count.  Let's track it as a separate issue?

Reply:
Sure.